src/xr.js

```diff
@@ -38,6 +38,7 @@ const defaults = {
   xmlHttpRequest: () => new XMLHttpRequest(),
   promise: fn => new Promise(fn),
   withCredentials: false,
+  timeout: 0,
```

Comment:
Updated and rebased my code to set a default value, since it seemed a cleaner approach than `|| 0`.